tap_eloqua/sync.py

```diff
@@ -241,10 +243,14 @@ def update_current_stream(state, stream_name):
     state['current_stream'] = stream_name
     singer.write_state(state)
 
-def should_sync_stream(last_stream, selected_streams, stream_name):
-    if last_stream == stream_name or \
-       (last_stream is None and stream_name in selected_streams):
```

Comment:
Instead of relying on a global, couldn't this just check that `last_stream` is in `selected_streams`, and if not, set it to `None` with the same predicate on this side of the diff? Since strings are immutable, that should allow the logic to work the same in the other calling locations with no side effects.

Reply:
@dmosorast I'm not sure I follow. If we don't use a global, how would we mark that we've encountered the `last_stream`?

Reply:
Ohhh, I see. That's subtle, and I guess speaks to why globals make me uncomfortable. In that case, it might be better to replace selected streams, with a `stream_info` map with `{selected_streams: [...], last_stream: "..."}` as a reference-style parameter.

Reply:
Then it can be modified directly without the all-caps variable peppered throughout